src/axl.c

```diff
@@ -610,6 +657,74 @@ int AXL_Dispatch (int id)
     return rc;
 }
 
+/*
+ * Given a path with an AXL temporary extension, allocate an return a new
+ * string with the extension removed.  Also, if extra is specified, return
+ * a pointer to the offset in 'path_with_extension' where the 'extra' field
+ * is.
+ *
+ * Returns the new allocated path string on success, or NULL on error.
+ */
+static char * axl_remove_extension(char *path_with_extension, char **extra)
```

Comment:
Does this need to account for the IBM handle if one is attached?

Reply:
Correct, it accounts for the transfer handle

Reply:
Oh, I see.  My first impression was that extra was an input parameter rather than an output value.